src/HieDb/Types.hs

```diff
@@ -31,8 +32,40 @@ import Database.SQLite.Simple.FromField
 import qualified Text.ParserCombinators.ReadP as R
 
 import HieDb.Compat
+import Control.Monad (void)
+
+data HieDb = HieDb
+  { getConn :: !Connection
+  , insertModsStatement :: !(StatementFor HieModuleRow)
+  , insertRefsStatement :: !(StatementFor RefRow)
+  , insertDeclsStatement :: !(StatementFor DeclRow)
+  , insertImportsStatement :: !(StatementFor ImportRow)
+  , insertDefsStatement :: !(StatementFor DefRow)
+  , insertExportsStatement :: !(StatementFor ExportRow)
+  , insertTyperefsStatement :: !(StatementFor TypeRef)
+  , insertTypenamesStatement :: !(StatementFor (OccName, ModuleName, Unit))
+  , queryTypenamesStatement :: !(StatementFor (OccName, ModuleName, Unit))
+  , deleteInternalTablesStatement :: !(Only FilePath -> IO ())
+  }
```

Comment:
I think we should document here the purpose of `StatementFor` and mention the performance benefits